packages/frontend/src/lib/creatorReadHooks.ts

```diff
@@ -132,7 +140,123 @@ export const useClaimableAmount = ({ beneficiary }: { beneficiary?: string }) =>
+  const [isLoading, setIsLoading] = useState(false)
+  const { contractsChain, contractsChainId, contracts } = useDeployments()
+  const [poolAddresses, setPoolAddresses] = useState<{
+    [_: string]: string | false
+  }>({})
+
+  const refetch = async (chainId: number | undefined) => {
+    if (!beneficiary || !contracts || !chainId) return
+    setIsLoading(true)
+    setPoolAddresses((prev) => ({
+      ...prev,
+      [chainId]: undefined,
+    }))
+    let poolAddress: string | false
+    try {
+      const provider = getDefaultProvider(env.rpcUrls[chainId as keyof typeof env.rpcUrls])
+      const factoryContract = YieldGate__factory.connect(contracts.YieldGate.address, provider)
+      poolAddress = await factoryContract.beneficiaryPools(beneficiary)
+      if (!poolAddress || poolAddress === ethers.constants.AddressZero) {
+        poolAddress = false
+      }
+    } catch (e) {
+      // do nothing
+    }
+    setPoolAddresses((prev) => ({
+      ...prev,
+      [chainId]: poolAddress,
+    }))
+    setIsLoading(false)
+  }
+  useEffect(() => {
+    refetch(contractsChainId)
+  }, [beneficiary, contractsChainId])
+
+  return {
+    isLoading,
+    poolAddresses,
+    ...(contractsChainId ? { poolAddress: poolAddresses[contractsChainId] } : {}),
+    contractChain: contractsChain,
+    contractChainId: contractsChainId,
+    refetch: async () => {
+      refetch(contractsChainId)
+    },
+  }
+}
+
+/**
+ * Returns the pool parameters (minAmount, minDurationDays) for a given `poolAddress`
+ */
+export const usePoolParams = ({ poolAddress }: { poolAddress?: string | false }) => {
+  const [isLoading, setIsLoading] = useState(false)
+  const { contractsChain, contractsChainId, contracts } = useDeployments()
+  const [minAmounts, setMinAmounts] = useState<{
+    [_: string]: string
+  }>({})
+  const [minDurationsDays, setMinDurationsDays] = useState<{
+    [_: string]: number
+  }>({})
+
+  const refetch = async (chainId: number | undefined) => {
+    if (!contracts || !chainId) return
+    if (!poolAddress) {
+      setMinAmounts((prev) => ({
+        ...prev,
+        [chainId]: undefined,
+      }))
+      setMinDurationsDays((prev) => ({
+        ...prev,
+        [chainId]: undefined,
+      }))
+      return
+    }
+
+    setIsLoading(true)
+    let minAmount = BigNumber.from(0)
+    let minDurationSeconds = BigNumber.from(0)
+    try {
+      const provider = getDefaultProvider(env.rpcUrls[chainId as keyof typeof env.rpcUrls])
+      const poolContract = BeneficiaryPool__factory.connect(poolAddress, provider)
+      minAmount = await poolContract.minAmount()
+      minDurationSeconds = await poolContract.minDuration()
+    } catch (e) {
+      // do nothing
+    }
+    setMinAmounts((prev) => ({
+      ...prev,
+      [chainId]: parseFloat(formatEther(minAmount) || '0.0'),
+    }))
+    setMinDurationsDays((prev) => ({
+      ...prev,
+      [chainId]: parseInt(minDurationSeconds.div(24 * 60 * 60).toString() || `0`),
```

Comment:
could also use `dayjs.duration(minDurationSeconds, 's').asDays()` here

Reply:
Switched to it in https://github.com/yieldgate/yieldgate/pull/56/commits/31c6808efb77df38fa330644fde5dde4894f1dc8